src/js/renderers/ArrayLayoutRenderer.js

```diff
@@ -189,7 +189,7 @@ export const ArrayControl = ( {
 						schema={ schema }
 						label={ label } 
 						path={ path }
-						route={ `${ route }/${ data.length }` } 
+						route={ `${ route }/${ data?.length || 0 }` } 
```

Comment:
The `addItem` button is outside of the `data` invalid check, therefore `data.length` is invalid when data is null or undefined